File: Data Structures & Algorithms/eating-bananas/submission-0.java
class Solution {
    public int minEatingSpeed(int[] piles, int h) {
        int left = 1;
        int right = 1;
        int ans = 1;

        for (int pile : piles) {
            right = Math.max(right, pile);
        }

        while (left <= right) {
            int mid = left + (right - left) / 2;
            int hours = 0;

            for (int pile : piles) {
                if (pile % mid == 0) {
                    hours += pile/mid;
                } else hours += (pile/mid) + 1;
            }

            if (hours <= h) {
                ans = mid;
                right = mid - 1;
            } else {
                left = mid + 1;
            }
        }

        return ans;
    }
}

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-0.go
func characterReplacement(s string, k int) int {
	mp := make(map[rune]int)
	str := []rune(s)
	left, ans, maxFreq := 0, 0, 0

	for right :=0; right < len(str); right++ {
		mp[str[right]]++
		maxFreq = max(maxFreq, mp[str[right]])

		windowSize := right - left + 1
		need := windowSize - maxFreq

		if need > k {
			mp[str[left]]--
			left++

			windowSize = right - left + 1
			need = windowSize - maxFreq
		}

		ans = max (ans, right - left +1)
	}

	return ans
}
